### ISSUE TITLE
Asset Processor ignores the exclude and include platforms when processing assets

### ISSUE THREAD

**Issue by LB-TomaszSerwanski** [o3de/o3de#5302]
**Describe the bug**
The _exclude_ and _include_ tags and platforms are ignored by the Asset Processor. This results in the _excluded_ (as well as the not _included_ ones) platforms being processed by the Asset Processor. 

Please note that the issue is also active on Linux.

Please refer to the attached screenshots for more details.

**Steps to reproduce**
1. Create a new folder in the project folder (i.e. <project_root>/Assets/foo).
2. Insert some assets into the newly created folder.
3. Add the newly created project folder to the <engine_root>/Registry/AssetProcessorPlatformConfig.setreg file, with the excluded Android platform (refer to the attached AssetProcessorPlatform_ScanFolder.png).
4. Uncomment the Android platform in the Platforms section of the same file (refer to the attached AssetProcessorPlatform_Platforms.png).
5. Save the file.
6. Run AssetProcessor.exe

**Expected behavior**
Assets for the excluded platforms are not processed.

**Actual behavior**
Assets for the excluded platforms are processed.

**Assets required**
N/A

**Screenshots/Video**

![AssetProcessor](https://user-images.githubusercontent.com/86953108/140314772-b573eed9-4e01-4309-8a4f-85fa92cce518.png)
![AssetProcessorPlatform_Platforms](https://user-images.githubusercontent.com/86953108/140314776-4e83ba94-dc83-4a08-ac6d-4e6831a0945d.png)
![AssetProcessorPlatform_ScanFolder](https://user-images.githubusercontent.com/86953108/140314777-3ffe167d-5252-498d-b5c0-748367dceeeb.png)

**Found in Branch**
Stabilization_2110 (d598a7c) 

**Desktop/Device:**
 - Device: PC
 - OS: Windows
 - Version: 10
 - CPU: Amd Ryzen 5 3600
 - GPU: Nvidia RTX 2060 SUPER
 - Memory: 16GB

**Comment by jackalbe** [comment]
Have you tried to remove the blank spaces between the platform identifiers? Such as:
```
"exclude":"android,es3,pc",
```

**Comment by LB-TomaszSerwanski** [comment]
Hi @jackalbe, sorry for the delayed reply. Yes, even with the blank spaces removed, the result is the same (I have also tried to exclude a single platform, which also did not work).

**Comment by jackalbe** [comment]
The issue is that the ""@PROJECTROOT@" root folder will continue to include all the files even those there is another scan folder inside the root folder that was excluded. If the project wants to include all the source assets in the same root folder, then the assets can be separated between the Shared assets, Windows only assets, and Android only assets. Then exclude the platform specific assets from the other platforms.

It is possible to have your assets broken up into different folders from the root like:
`
"ScanFolder Project/Assets/Shared": {
  "watch": "@PROJECTROOT@/Shared",
  "recursive": 1,
  "order": 0
},
"ScanFolder Assets/Platform/Android": {
  "watch": "@PROJECTROOT@/Platform/Android",
  "recursive": 1,
  "order": 50000,
  "exclude": "pc"
},
"ScanFolder Assets/Platform/Windows": {
  "watch": "@PROJECTROOT@/Platform/Windows",
  "recursive": 1,
  "order": 50000,
  "exclude": "android"
}
`

**Comment by LB-TomaszSerwanski** [comment]
@jackalbe you are correct, once I set it up this way, the _exclude_ tag works correctly, thank you. Verified on Stabilization_2110 (76b7e73).

**Comment by alexmontAmazon** [comment]
@LB-TomaszSerwanski , @jackalbe , is this issue now resolved?
